src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrayLength = this.length;
+    let result = '';
+
+    for (let i = 0; i < arrayLength; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+      } else {
+        result += this[i];
```

Comment:
Use explicit conversion to string here to ensure the correct type of the value - result += String(this[i]);